### ISSUE TITLE
Swift logger fails to write new log files until app relaunch if logs are cleared

### ISSUE THREAD

**Issue by jamilbk** [firezone/firezone#7569]
The Swift log module fails to recreate the files that are cleared by the `clearLogs` functionality.

This is not a new bug introduced during the development of #7071 -- it's been around for some time.

The fix here is likely to check if the file exists before each write, or use a write that returns an error if the file is unavailable.

**Comment by thomaseizinger** [comment]
in the Tauri client, we don't clear the file we are currently writing to for the same reason.

**Comment by thomaseizinger** [comment]
@jamilbk Does this need to be fixed before the 1.4.0 release?

**Comment by thomaseizinger** [comment]
Just hit this with a customer that thought clearing all logs will roll over to a new one. This affects connlib's logger as well.

**Comment by jamilbk** [comment]
Yeah, I don't think we'll be able to address the connlib logger as part of this issue. Perhaps an FFI call is needed to forcibly reset the file being written to?

**Comment by thomaseizinger** [comment]
> in the Tauri client, we don't clear the file we are currently writing to for the same reason.

Maybe just replicate this functionality in the Apple client?

**Comment by jamilbk** [comment]
> > in the Tauri client, we don't clear the file we are currently writing to for the same reason.
> 
> Maybe just replicate this functionality in the Apple client?

Ah, I was actually to suggest we fix that too. I've been on two calls now where I watched the user become confused when they clicked "clear logs" and it still showed "2 files, XX KB".

**Comment by jamilbk** [comment]
Not the biggest UX issue in the world, but I guess not super hard to fix either?

**Comment by thomaseizinger** [comment]
> > > in the Tauri client, we don't clear the file we are currently writing to for the same reason.
> > 
> > 
> > Maybe just replicate this functionality in the Apple client?
> 
> Ah, I was actually to suggest we fix that too. I've been on two calls now where I watched the user become confused when they clicked "clear logs" and it still showed "2 files, XX KB".

We can change the button text / description to "Delete old logs" or something like that. Rolling over is also possible but requires some work across all layers to pass through a "rollover" command.

**Comment by thomaseizinger** [comment]
Fixed in #9597.
